src/test/java/uk/gov/hmcts/reform/sendletter/tasks/SerialTaskRunnerLockingTest.java

```diff
@@ -0,0 +1,187 @@
+package uk.gov.hmcts.reform.sendletter.tasks;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import javax.sql.DataSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.Matchers.startsWith;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.only;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SerialTaskRunnerLockingTest {
+
+    @Mock
+    private DataSource source;
+    @Mock
+    private Connection connection;
+    @Mock
+    private Statement statement;
+    @Mock
+    private Runnable task;
+    private SerialTaskRunner taskRunner;
+
+    @Before
+    public void setup() throws SQLException {
+        when(connection.createStatement()).thenReturn(statement);
+        when(source.getConnection()).thenReturn(connection);
+
+        taskRunner = SerialTaskRunner.get(source);
+    }
+
+    @Test
+    public void runs_task_after_successful_locking() throws SQLException {
+        setupSuccessfulLocking();
+        setupSuccessfulUnlocking();
+
+        taskRunner.tryRun(1, task);
+
+        verify(task, only()).run();
+    }
+
+    @Test
+    public void runs_task_after_failed_unlocking() throws SQLException {
+        setupSuccessfulLocking();
+        setupFailedUnlocking();
+
+        taskRunner.tryRun(1, task);
+
+        verify(task, only()).run();
+    }
+
+    @Test
+    public void does_not_run_task_after_failed_locking() throws SQLException {
+        setupFailedLocking();
+        setupFailedUnlocking();
+
+        taskRunner.tryRun(1, task);
+
+        verify(task, never()).run();
+    }
+
+    @Test
+    public void does_not_run_task_after_failed_locking_and_failed_unlocking() throws SQLException {
+        setupFailedLocking();
+        setupSuccessfulUnlocking();
```

Comment:
either wrong method call or test name (unlocking successful vs failed)

Reply:
Damn, should've left these cryptic method names ;)